rust/stackable-cockpit/src/xfer/cache.rs

```diff
@@ -97,15 +109,14 @@ impl Cache {
                 let mut entries = fs::read_dir(base_path).await.context(CacheIoSnafu)?;
 
                 while let Some(entry) = entries.next_entry().await.context(CacheIoSnafu)? {
-                    files.push((
-                        entry.path(),
-                        entry
-                            .metadata()
-                            .await
-                            .context(CacheIoSnafu)?
-                            .modified()
-                            .context(CacheIoSnafu)?,
-                    ))
+                    let metadata = entry.metadata().await.context(CacheIoSnafu)?;
+
+                    // Skip the last-auto-purge file
+                    if metadata.is_file() && is_protected_file(entry.file_name())? {
+                        continue;
+                    }
+
+                    files.push((entry.path(), metadata.modified().context(CacheIoSnafu)?))
                 }
 
                 files.sort();
```

Comment:
Unrelated: does anything rely on this sorting? Why do we do it here rather than where that assumption exists?

Reply:
Removed in b51571d